src/Draco.Compiler/Internal/Binding/Binder_BoundExpression.cs

```diff
@@ -87,6 +102,16 @@ private BoundUnexpectedExpression TypeModuleExpression(UntypedModuleExpression m
     private BoundExpression TypeGlobalExpression(UntypedGlobalExpression global, ConstraintSolver constraints, DiagnosticBag diagnostics) =>
         new BoundGlobalExpression(global.Syntax, global.Global);
 
+    private BoundExpression TypeFieldExpression(UntypedFieldExpression field, ConstraintSolver constraints, DiagnosticBag diagnostics) =>
+        new BoundFieldExpression(field.Syntax, field.Reciever is null ? null : this.TypeExpression(field.Reciever, constraints, diagnostics), field.Field);
+
+    private BoundExpression TypePropertyGetExpression(UntypedPropertyGetExpression prop, ConstraintSolver constraints, DiagnosticBag diagnostics) =>
+        new BoundPropertyGetExpression(prop.Syntax, prop.Getter, prop.Receiver is null ? null : this.TypeExpression(prop.Receiver, constraints, diagnostics));
+
+    private BoundExpression TypeIndexGetExpression(UntypedIndexGetExpression index, ConstraintSolver constraints, DiagnosticBag diagnostics) =>
+        new BoundIndexGetExpression(index.Syntax, index.Getter.Result, this.TypeExpression(index.Receiver, constraints, diagnostics),
+            index.Indices.Select(x => this.TypeExpression(x, constraints, diagnostics)).ToImmutableArray());
```

Comment:
The implementations are perfectly fine, but please make these block bodied methods and factor out computed results. For ex. this:
```cs
new BoundPropertyGetExpression(prop.Syntax, prop.Getter, prop.Receiver is null ? null : this.TypeExpression(prop.Receiver, constraints, diagnostics));
```
would become this:
```cs
var receiver = prop.Receiver is null ? null : this.TypeExpression(prop.Receiver, constraints, diagnostics);
return new BoundPropertyGetExpression(prop.Syntax, prop.Getter, receiver );
```
The reason is easier debuggability, locals are very easy to read in the locals window

Reply:
And please do the same at the rest of the places in binders